packages/expo-contacts/ios/next/form/FormDelegate.swift

```diff
@@ -0,0 +1,159 @@
+  }
+  
+  private let delegate = ContactControllerDelegate()
+  private var presentingViewController: UIViewController?
+  private var contactPickerDelegate: ContactPickerControllerDelegate?
+  private var contactPickingPromise: Promise?
+  private var contactManipulationPromise: Promise?
+  
+  func presentEditForm(for contact: ContactNext, options: FormOptions?, promise: Promise) throws {
+    if contactManipulationPromise != nil {
+      throw ContactManipulationInProgressException()
+    }
+    guard let foundContact = contactRepository.getById(
+      id: contact.id,
+      keysToFetch: [CNContactViewController.descriptorForRequiredKeys()]
+    ) else {
+      return promise.reject(ContactNotFoundException(contact.id))
+    }
+    
+    var controller = ContactsViewController(for: foundContact)
+    try setControllerOptions(controller: &controller, options: options)
+    
+    
+    if let parent = appContext?.utilities?.currentViewController() {
+      let navController = UINavigationController(rootViewController: controller)
+      presentingViewController = navController
+      
+      controller.onViewDisappeared = {
+        promise.resolve()
+        self.contactManipulationPromise = nil
+      }
+      
+      contactManipulationPromise = promise
+      parent.present(navController, animated: options?.preventAnimation ?? false != true)
+    } else {
+      contactManipulationPromise = nil
+      promise.reject(MissingViewControllerException())
+    }
+  }
+  
+  func presentAddForm(contact: CNContact, options: FormOptions?, promise: Promise) throws {
+    if contactManipulationPromise != nil {
+      throw ContactManipulationInProgressException()
+    }
+    var controller = ContactsViewController(for: contact)
+    try setControllerOptions(controller: &controller, options: options)
+    
+    if let parent = appContext?.utilities?.currentViewController() {
+      let navController = UINavigationController(rootViewController: controller)
+      presentingViewController = navController
+      
+      controller.onViewDisappeared = {
+        promise.resolve()
+        self.contactManipulationPromise = nil
+      }
+      
+      contactManipulationPromise = promise
+      parent.present(navController, animated: options?.preventAnimation ?? false != true)
+    } else {
+      contactManipulationPromise = nil
+      promise.reject(MissingViewControllerException())
+    }
+  }
+  
+  func presentPicker(promise: Promise) throws {
+    if contactPickingPromise != nil {
+      throw ContactPickingInProgressException()
+    }
+
+    let pickerController = CNContactPickerViewController()
+
+    contactPickerDelegate = ContactPickerControllerDelegate(onContactPickingResultHandler: self)
+
+    pickerController.delegate = self.contactPickerDelegate
+
+    let currentController = appContext?.utilities?.currentViewController()
+
+    contactPickingPromise = promise
+
+    currentController?.present(pickerController, animated: true)
```

Comment:
```suggestion

    let pickerDelegate = ContactPickerControllerDelegate(onContactPickingResultHandler: self)

    let pickerController = CNContactPickerViewController()
    pickerController.delegate = pickerDelegate

    let currentController = appContext?.utilities?.currentViewController()
    currentController?.present(pickerController, animated: true)
    
    contactPickerDelegate = pickerDelegate
    contactPickingPromise = promise
```
Should we reject the promise if `currentController` is null?